### PR TITLE
Fix issue 668 -- added path handling in pspm_init and pspm_quit

### LINKED ISSUE #668: Path handling in `pspm_init` and `pspm_quit`
### Summary

PsPM path handling during startup is opaque. Some folders are put on the path and not removed. It is unclear whether all required folders are put on the path.

1. Collect all folders that need to be put on PsPM path on startup and list them here.
2. In `pspm_init` check which of these folders are already on the path, and store them in `settings`. 
3. Put all folders on the path.
4. in `pspm_quit`, remove those folders from the path that were not on the path when PsPM was started.

### PR DESCRIPTION
**Pull request by MadniAbdulWahab**
Fixes #668  .

Changes proposed in this pull request:
- Defined directories and added necessary paths in pspm_init.
- Removed the paths in pspm_quit which were added in pspm_init.

### CODE REVIEW

File: src/pspm_init.m [hunk 1]
```diff
@@ -41,10 +41,27 @@
...
-if ~isempty(settings) % initialise settings
-  settings = [];
+initial_paths = strsplit(path, pathsep);
+global settings;
+if isempty(settings)
+  settings = struct();
 end
+% Define the directories PsPM will need
+pspm_root = fileparts(which('pspm_init'));
+required_folders = {'backroom', 'ext', 'Import', 'pspm_cfg'};
```
Comment (teddychao) [inline src/pspm_init.m:51]: Perhaps it is useful to ask @dominikbach if `backroom` is expected here. I remember `backroom` is expected to be excluded in a previous commit but I am not fully certain.

**Comment by teddychao** [comment]
I think the PsPM folders in this list shall be excluded into the path list for being added to MATLAB.
image
It is recommended to only add those folders instead of folders and their subfolders.

**Review by teddychao — CHANGES_REQUESTED** [review]

**Comment by teddychao** [comment]
This is now finished.

Before starting PsPM
image

When PsPM is running
image

After quitting PsPM
image

**Comment by teddychao** [comment]
### Discussion 01.07.2024

The key tasks
1. The subfolders may cause issues if they are all added to MATLAB. The essential subfolders need to be identified. Other folders which may cause conflicts shall be excluded to the subfolder list (the essential subfolders).
2. SPM may cause incompatibility when users run PsPM with SPM path added. This shall be solved by adding a warning in `pspm_init` and asking users to remove PsPM.
3. The subfolder list shall be updated at the beginning of `pspm_init`. This needs to be managed by PsPM automatically without asking users to do that.

**Comment by dominikbach** [comment]
To add on the previous points:
1. If SPM or Matlabbatch installations (outside PsPM folders) are on the path already, interrupt start up and give an error.
2. If any conflicting PsPM subfolders are on the path, interrupt startup and give an error.
3. Add root (src) folder to list of essential folders to be added to the path on startup.

**Comment by teddychao** [comment]
Examples

Warning dialog
image

Error
image

**Comment by teddychao** [comment]
Hi @dominikbach This has been completed. Please may I ask you to test and let me know if you are happy with this version. Thanks.

**Review by teddychao — APPROVED** [review]

**Pull request merged**